### PR TITLE
refactor: making IsBound a public function

### PR DESCRIPTION
**Pull request by seantking**
## Description

Making `IsBound` 05-port keeper fn public. 

closes: #288 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by seantking** [comment]
@colin-axner should I make a changelog for this? Under improvements?

**Comment by colin-axner** [comment]
> @colin-axner should I make a changelog for this? Under improvements?

Yes definitely!

**Review by colin-axner — COMMENTED** [review]
Should we add a test?

Would like an ACK from @AdityaSripal as I don't have the context as to why this was private to begin with

**Review by AdityaSripal — APPROVED** [review]

**Comment by seantking** [comment]
> Should we add a test?

Done :+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (modules/light-clients/07-tendermint) [\#125](https://github.com/cosmos/ibc-go/pull/125) Implement efficient iteration of consensus states and pruning of earliest expired consensus state on UpdateClient.
 * (modules/light-clients/07-tendermint) [\#141](https://github.com/cosmos/ibc-go/pull/141) Return early in case there's a duplicate update call to save Gas.
 * (modules/core/ante) [\#235](https://github.com/cosmos/ibc-go/pull/235) Introduces a new IBC Antedecorator that will reject transactions that only contain redundant packet messages (and accompany UpdateClient msgs). This will prevent relayers from wasting fees by submitting messages for packets that have already been processed by previous relayer(s). The Antedecorator is only applied on CheckTx and RecheckTx and is therefore optional for each node.
+* (modules/core/05-port) [\#288](https://github.com/cosmos/ibc-go/issues/288) Making the 05-port keeper function IsBound public. The IsBound function checks if the provided portID is already binded to a module.
```
Comment (colin-axner) [inline CHANGELOG.md:108]: This should go in the unreleased section

**Review by colin-axner — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (seantking) [reply CHANGELOG.md]: thank you :+1:

**Pull request merged**